Deal with remaining disabled tests

canelhasmateus opened the pull request.

canelhasmateus commented on tests/assign/toverload_asgn2.nim:
```diff
@@ -1,7 +1,7 @@
   output: '''i value 88
 2aa'''
-  disabled: "true"
+    disabled: "true"
```
Disabled just for CI's sake around 2018-04. 
at the timeline, very near fixes to issues #7637, #6393, altough doesn't seem related to any of them.
This test doesn't work, but seems fixable. 
Running it gives the log:
		
    Expected:
    i value 88
    2aa

    Gotten:
    i value 88
    0aa

saem replied:
Yeah, I think this might be covered by the spec bits @haxscramper is looking at now.

haxscramper replied:
No, I don't think it is related, or at least I don't see it immediately - how `=` is handled is another can of worms that I should probably check as well.

saem replied:
Just read this code again and the imported module, v is a let, so it's not a var so the assignment overload never kicks off. It will for `x`, but I'm not sure if the `result` in the `passAround` call is being caught by `=`... that might well be a bug.

Mark it as a known issue, description could be something like "possible issue where `result` is not treated as `var Concrete` in `passAround`", or something like that.

canelhasmateus commented on tests/bind/tdatabind.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-  disabled: true
+    disabled: true
```
I believe this should be deleted.

saem replied:
Exterminate!

canelhasmateus commented on tests/closure/texplicit_dummy_closure.nim:
```diff
@@ -1,5 +1,5 @@
+    disabled: true
 """
 
 # This is a regression of the new lambda lifting; detected by Aporia
```
I believe this should be deleted.

saem replied:
Exterminate!

canelhasmateus commented on tests/cpp/tdont_init_instantiation.nim:
```diff
@@ -1,7 +1,7 @@
   targets: "cpp"
   output: ''''''
-  disabled: true
+    disabled: true
```
References  https://github.com/nim-lang/Nim/issues/5140
Disabled around 2016-12, for CI sake, it seems.

saem replied:
I'm not sure what this even means.

Ping @haxscramper for some insight, should we just delete it or keep it disabled?

haxscramper replied:
If this does not work properly, we can just put a "known issue" on the most Cxx interop - I mean, if nim clobbers the init data of the class constructor, it is really bad.

canelhasmateus replied:
Added a knownIssue, pointing to the issue it refered to when created (#5140)

canelhasmateus commented on tests/effects/toutparam.nim:
```diff
@@ -2,7 +2,7 @@ discard """
   errormsg: "use explicit initialization of 'x' for clarity [Uninit]"
   line: 24
-  disabled: "true"
+    disabled: "true"
```
created disabled around 2020-06 in a commit that references https://github.com/nim-lang/Nim/issues/14521

Running it gives:

			Expected:
			use explicit initialization of 'x' for clarity [Uninit]

			Gotten:
			use explicit initialization of 'tmp' for clarity [Uninit]

saem replied:
I figure a known issue is fine for this.

canelhasmateus replied:
Do we have a issue number?

saem replied:
no, but a textual description is fine.

I want to be careful before we commit to this being a "for real" known issue, because as we're working through the spec things are still settling, so something like this we'll end up revisiting.

canelhasmateus replied:
How should a textual description be done? in the known issue field?

saem replied:
Yeah, I would put the text in the known issue field.

canelhasmateus commented on tests/compiler/tdebugutils.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
   joinable: false
-  disabled: true
+    disabled: true
```
was born disabled. I guess it is used only for demonstration purposes? 
		from my current understanding, these type of tests should be coalesced into the specs?

saem replied:
Or a debugging guide or straight into the module `debugutils`, this is dumb, think you could document it therein?

canelhasmateus replied:
i believe @quantimnot is working on [someting among those lines](https://github.com/saem/vscode-nim/issues/60#issuecomment-958635999), but otherwise, i surely can.

saem replied:
different debugging, this is glorified echo/print based debugging.

canelhasmateus replied:
I'll be keeping this as disabled for now. I will probably get around to this documentation soon&trade;

canelhasmateus commented on tests/generics/toverloading_typedesc.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
   exitcode: 0
-  disabled: '''true'''
+    disabled: '''true'''
```
Crreated around 2015-06 in a commit that references https://github.com/nim-lang/Nim/issues/3022
Disabled around 2017-11 for CI sake?
Working in my pc.

saem replied:
Enable

canelhasmateus commented on tests/global/tglobal.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
   output: "in globalaux2: 10\ntotal globals: 2\nint value: 100\nstring value: second"
-  disabled: "true"
+    disabled: "true"
```
Disabled around 2015-03, for CI sake?
Working on my pc.

saem replied:
Enable

canelhasmateus commented on tests/iter/titerable.nim:
```diff
@@ -6,7 +6,7 @@ discard """
 12
 '''
-  disabled: "true"
+    disabled: "true"
```
Created in 2014-03 in a commit that references https://github.com/nim-lang/Nim/issues/966
Disabled around 2015-11 citing 'lambda lifting support for iterToProc plugin'

saem replied:
Known issue

canelhasmateus replied:
Do we have a issue number?

saem replied:
nope, I'd go with textual description, lambda lifting and iterators are likley to go through a bit spec definition and this will either get folded in and it's a good signal to know whether we fixed something by happenstance or not.

canelhasmateus commented on tests/iter/tscheduler.nim:
```diff
@@ -7,7 +7,7 @@ a2 8
 a2 4
 a2 2'''
-  disabled: "true"
+    disabled: "true"
```
Disabled around 2015-07, citing dubiousness ( flakiness? )
Working in my pc.

saem replied:
Coro, remove

canelhasmateus commented on tests/method/tmultim.nim:
```diff
@@ -10,7 +10,7 @@ collide: thing, unit |
 '''
   joinable: false
-  disabled: true
+    disabled: true
```
Disabled iaround 2020-03 in a commit that references https://github.com/nim-lang/Nim/issues/13803, for CI sake?
Working in my pc.

saem replied:
Enable

canelhasmateus commented on tests/method/tmultimjs.nim:
```diff
@@ -4,7 +4,7 @@ discard """
 hello
 '''
-  disabled: true
+    disabled: true
```
Disabled iaround 2020-03 in a commit that references https://github.com/nim-lang/Nim/issues/13803, for CI sake?
Working in my pc.

saem replied:
Enable

canelhasmateus commented on tests/misc/tdllvar.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-disabled: true
+    disabled: true
```
Created around 2014-04 in a commit that references https://github.com/nim-lang/Nim/issues/1075
Disabled around 2018-11 for CI sake?

saem replied:
Disable, I guess, not sure

canelhasmateus replied:
Keeping this disabled for now.

canelhasmateus commented on tests/misc/tgenconstraints.nim:
```diff
@@ -2,7 +2,7 @@ discard """
   file: "tgenconstraints.nim"
   line: 25
-  disabled: true
+    disabled: true
```
Created disabled around 2014-04 in a commit that references https://github.com/nim-lang/Nim/issues/1075
Running gives the output:

	Expected:
	cannot instantiate T2

	Gotten:
	undeclared identifier: 'TNumber'

saem replied:
This test doesn't make sense, there isn't a TNumber, expectation seems wrong.

canelhasmateus replied:
I'll take this as a delete.

canelhasmateus commented on tests/misc/tmissingnilcheck.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-  disabled: true
+    disabled: true
```
Created disabled around 2014-04 in a commit that references https://github.com/nim-lang/Nim/issues/1075

saem replied:
How does it fail?

canelhasmateus replied:
```
Expected:
exitcode: 0

Gotten:
exitcode: 1

Output:
Traceback (most recent call last)
tmissingnilcheck.nim(20) tmissingnilcheck
tmissingnilcheck.nim(18) newConnection
gc.nim(280)              asgnRef
SIGSEGV: Illegal storage access. (Attempt to read from nil?)

```

saem replied:
I really don't get how this is a missing nil check test... weird.

unless the history provides some clues I would consider deleting it.

canelhasmateus replied:
None whatsover. No meaningful changes since its inception.

saem replied:
Ick, let's delete it.

canelhasmateus commented on tests/misc/tunsignedcomp.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
   output: ''''''
-  disabled: "true"
+    disabled: "true"
```
Created around 2015-01
Disabled around 2015-01, in a commit that says "cleaned up unsigned tests"

saem replied:
This doesn't work?

Pretty sure we have similar tests in the spec

canelhasmateus replied:
This seems like basic basic functionality, so i would expect there to be similar or redundant tests .
For completeness, the log:
```
FAIL: tests/misc/tunsignedcomp.nim c                               ( 0.45 sec)
Test "tests\misc\tunsignedcomp.nim" in category "misc"
Failure: reNimcCrash
debugInfo:  compiler exit code was 0 but some Error's were found.
$ C:\Users\Mateus\Desktop\workspace\nimskull\bin\nim.exe c --hints:on -d:testing --clearNimblePath --nimblePath:build/deps/pkgs   --nimCache:nimcache\
tests\misc\tunsignedcomp.nim_4a8a08f09d37b73795649038408b5f33  tests\misc\tunsignedcomp.nim
Hint: used config file 'C:\Users\Mateus\Desktop\workspace\nimskull\config\nim.cfg' [Conf]
Hint: used config file 'C:\Users\Mateus\Desktop\workspace\nimskull\config\config.nims' [Conf]
Hint: used config file 'C:\Users\Mateus\Desktop\workspace\nimskull\tests\config.nims' [Conf]
tunsignedcomp.nim(13, 16) Error: type mismatch: got <uint8, int8>
but expected one of:
proc `<`(x, y: uint): bool
  first type mismatch at position: 2
  required type for y: uint
  but expression '20'i8' is of type: int8
proc `<`(x, y: uint16): bool
  first type mismatch at position: 2
  required type for y: uint16
  but expression '20'i8' is of type: int8
proc `<`(x, y: uint32): bool
  first type mismatch at position: 2
  required type for y: uint32
  but expression '20'i8' is of type: int8
proc `<`(x, y: uint64): bool
  first type mismatch at position: 2
  required type for y: uint64
  but expression '20'i8' is of type: int8
proc `<`(x, y: uint8): bool
  first type mismatch at position: 2
  required type for y: uint8
  but expression '20'i8' is of type: int8
16 other mismatching symbols have been suppressed; compile with --showAllMismatches:on to see them

expression: 10'u8 < 20'i8
```

saem replied:
yeah, seriously... this test just doesn't make sense, an easy fix would be to remove this particular failing line because I"m not sure if supporting that comparison even makes sense.

canelhasmateus commented on tests/modules/tnotuniquename2.nim:
```diff
@@ -1,7 +1,7 @@
   errormsg: "module names need to be unique per Nimble package"
   file: "tnotuniquename/mnotuniquename.nim"
-  disabled: "true"
+    disabled: "true"
```
Disabled around 2019-04 in a commit that references https://github.com/nim-lang/Nim/issues/11064
Seems to be testing removed behaviour. Probably should be removed.

canelhasmateus commented on tests/modules/trecmod.nim:
```diff
@@ -2,7 +2,7 @@ discard """
   file: "mrecmod.nim"
   line: 1
-  disabled: true
+    disabled: true
```
not sure about this one.

saem replied:
it should simply error out... I don't get why it's disabled.

canelhasmateus replied:
Created at 2008. 
Filled with something around 2016.
It got disabled the very next day, 
in a commit with message  `recursive modules are only detected to improve error messages`
running it gives:
```Failure: reMsgsDiffer
Expected:
recursive module dependency detected

Gotten:

```

saem replied:
Let's delete it.

canelhasmateus commented on tests/navigator/tincludefile.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-  disabled: true # this test is flakey with testament all
+    disabled: true # this test is flakey with testament all
```
Created around 2021-04 in a commit that references https://github.com/nim-lang/Nim/issues/17784
Disabled around 2021-07 due to flakiness.
probably should be removed.

saem replied:
yeah... we're not going to bother with this navigator stuff, let's kill it.

canelhasmateus commented on tests/objects/tobjpragma.nim:
```diff
@@ -8,7 +8,7 @@ discard """
 3
 '''
-disabled: "true"
+    disabled: "true"
```
Disabled around 2015-03 for CI sake
There seems to be a ton of issues referencing this:
https://github.com/nim-lang/Nim/issues/15752
https://github.com/nim-lang/Nim/issues/9668
https://github.com/nim-lang/Nim/issues/10082
https://github.com/nim-lang/Nim/pull/10227
https://github.com/nim-lang/Nim/issues/4763

seems to have been fixed on https://github.com/nim-lang/Nim/pull/10227

working on my pc

saem replied:
Enable and let's see what happens.

canelhasmateus commented on tests/parallel/t5000.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
   output: '''50005000'''
-  disabled: "true"
+    disabled: "true"
```
Created around 2014-11 in a commit that references https://github.com/nim-lang/Nim/issues/1646
Disabled around 2014-12.
Working on my pc, but probably flaky
what is this even testing, though
Prob. delete

saem replied:
Exterminate!

Seriously, I just can't be bothered with taking on fixing those libs right now.

canelhasmateus commented on tests/parallel/treadafterwrite.nim:
```diff
@@ -1,7 +1,7 @@
   errormsg: "'foo' not disjoint from 'foo'"
   line: 23
-  disabled: "true"
+    disabled: "true"
```
Created disabled around 2014-12 in a commit that references https://github.com/nim-lang/Nim/issues/1597

saem replied:
Does it work?

If not, then delete.

canelhasmateus replied:
For completeness: 

```
FAIL: tests/parallel/treadafterwrite.nim c                         ( 3.22 sec)
Test "tests\parallel\treadafterwrite.nim" in category "parallel"
Failure: reMsgsDiffer
Expected:
'foo' not disjoint from 'foo'

Gotten:



```
Yes, the 'gotten' field  is blank.

canelhasmateus commented on tests/pragmas/tpragmas_misc.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-  disabled: true
+    disabled: true
```
Created around 2020-05 in a commit that references https://github.com/nim-lang/Nim/issues/14443
Changed in a commit that references https://github.com/nim-lang/Nim/issues/15920
Disabled around 2021-10 for CI sake

saem replied:
Yeah, this is a known issue, not exactly sure it should work but pretty sure, pragmas might be still a bit under specified.

haxscramper replied:
"a bit" is certainly the word here

`let a {.foo.} = 1` should probably work, but I would have to look through this test in details to see whether expectations make sense. Ideally it could be split into stuff that works, and `knownIssue: true` file, so we continued to test parts that are currently working

canelhasmateus replied:
Keeping this as disabled for now.

canelhasmateus commented on tests/system/t7894.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-disabled: true
+    disabled: true
```
Created around 2018-08 in a commit that references https://github.com/nim-lang/Nim/issues/7894 and https://github.com/nim-lang/Nim/issues/8496
 Disabled on x32 around 2019-02 due to flakiness on CI - references https://github.com/nim-lang/Nim/issues/10608
 Seems to run fine. Re-enable?

saem replied:
This is a silly brute force test that'll make CI grumpy.

Should be a way to launch the test with limited memory to see if we can trigger the failure consistently.

canelhasmateus replied:
Keeping this disabled for now.

canelhasmateus commented on tests/system/tio.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
 outputsub: ""
-disabled: true
+    disabled: true
```
Disabled around 2011-12 because of infinite blocking, reenabled shortly after
References https://github.com/nim-lang/Nim/issues/7121
Disabled around 2018-11 due to 'just causing problems'

saem replied:
These ii tests really half-baked.

Exterminate!

canelhasmateus commented on tests/template/tmodulealias.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-  disabled: true
+    disabled: true
```
Disabled around 2011-05 in a commit that references #30 and #26
probably delete, 10 years already.

saem replied:
Exterminate!

canelhasmateus commented on tests/typerel/tno_gcmem_in_shared.nim:
```diff
@@ -1,7 +1,7 @@
   errormsg: "shared memory may not refer to GC'ed thread local memory"
   line: 14
-  disabled: true
+    disabled: true
```
Disabled around  2014, back when 'shared' was a keyword.

saem replied:
Exterminate!

canelhasmateus commented on tests/typerel/trectuple.nim:
```diff
@@ -1,7 +1,7 @@
   errormsg: "illegal recursion in type 'TNode'"
   line: 8
-  disabled: true
+    disabled: true
```
Disabled around 2014 in a commit that 'tuple field names are ignored'

saem replied:
Does it work?

canelhasmateus replied:
Currently errors, but i believe its easily fixable:

```Failure: reLinesDiffer
Expected:
8:0

Gotten:
9:13
```

saem replied:
Does 8:0 even make sense, not saying 9:13 is much better, maybe slightly... they're both silly, just degrees.

I'd say update the line number and bring it back?

canelhasmateus commented on tests/typerel/trectype.nim:
```diff
@@ -1,6 +1,6 @@
 discard """
   errormsg: "internal error: cannot generate C type for: PA"
-  disabled: true
+    disabled: true
```
Disabled around 2016-12.
This seems to correctly validate errors in recursive types, just that the message changed over time.
Running it gives:
	
			Expected:
			internal error: cannot generate C type for: PA

			Gotten:
			illegal recursion in type 'PA'
			
seems fixable

saem replied:
Yeah, let's migrate to spec.

canelhasmateus commented on tests/types/tisop.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-  disabled: true
+    disabled: true
```
Disabled around 2014 in a commit that 'tuple field names are ignored'

saem replied:
Might already be in the spec, and what isn't can either be added or discarded.

canelhasmateus commented:
@saem moved that info into a more digestible form here

saem reviewed (changes requested):
Most of the way through review, but here are the interim items.

saem commented on tests/stdlib/tparsopt.nim:
```diff
@@ -1,5 +1,5 @@
 discard """
-disabled: true
+    disabled: true
```
Known issue, design of parse options is dumb

canelhasmateus replied:
Do we have a issue number?

saem replied:
Don't want to track an issue for this, currently the standard library is mostly out of scope unless required to unblock compiler work. The noise and fielding PRs in that way is more work than it's worth right now.

saem reviewed (changes requested):
Wrapped up this pass

canelhasmateus commented:
tdebugutils.nim was deleted in another , unrelated commit. 

Remaining issues are:
t7894.nim, which makes CI grumpy.
tdllvar.nim
tisop.nim, which probably is covered by some other specs.
tpragmas_misc.nim

canelhasmateus commented:
I'll create a comprehensive commit message, detailing the changes and reasonings to each file, and this should be ready to merge, unless we can deal with the remaining problematic tests. 
If so, we can for sure remove the 'disabled' keyword from the testament spec.
Sorry about the delay, work got me by the balls.

canelhasmateus committed:
This is the second commit of an ongoing effort to remove cruft from the test suite,
and remove the 'disabled' keyword from the testament spec.

In the first commit, we added the new knownIssue key to the testament spec,
while also adding knownIssues to many of the tests for which we could attribute an issue.

In this commit, we almost finish the task of removing the 'disabled' keyword
from the remaining tests.

Below are the changes and reasonings behind them:

Removed Due to old or removed features:

    tdatabind.nim
    texplicit_dummy_closure.nim
    tgenconstraints.nim
    tincludefile.nim
    tmodulealias.nim
    tno_gcmem_in_shared.nim
    tnotuniquename2.nim
    tscheduler.nim

Removed due to being Half Baked or Non-sensical:

    tio.nim
    tmissingnilcheck.nim
    treadafterwrite.nim
    trecmod.nim
    tunsignedcomp.nim

Removed due to not being worth the trouble:

    t5000.nim

Removed due to being redundant:

    tisop.nim

Keeping disabled, due to being unsure:

    t7894.nim - Works, but flaky, and makes CI servers grumpy.
    tdllvar.nim
    tpragmas_misc.nim

Added knownIssues - Mostly textual descriptions, due to not having a explicit issues:

    tdont_init_instantiation.nim
    titerable.nim
    toutparam.nim
    toverload_asgn2.nim
    tparsopt.nim

Re-enabled:

    tglobal.nim
    tmultim.nim
    tmultimjs.nim
    tobjpragma.nim
    toverloading_typedesc.nim

Fixed

    trectuple.nim: Fixed the line number.
    trectype.nim : Fixed to the spec.

saem reviewed (approved):
bors r+

🎉
